pkg/component/etcd/etcd/crd.go

```diff
@@ -9,93 +9,106 @@ import (
 	_ "embed"
 	"fmt"
 
-	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/Masterminds/semver/v3"
+	druidcorev1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	druidcorecrds "github.com/gardener/etcd-druid/api/core/v1alpha1/crds"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/component"
 	"github.com/gardener/gardener/pkg/utils/flow"
 	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
 )
 
-var (
-	//go:embed crds/templates/crd-druid.gardener.cloud_etcds.yaml
-	// CRD holds the etcd custom resource definition template
-	CRD string
-	//go:embed crds/templates/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml
-	etcdCopyBackupsTaskCRD string
+type crd struct {
+	client       client.Client
+	applier      kubernetes.Applier
+	crdResources map[string]string
+}
 
-	etcdCRDName                = "etcds.druid.gardener.cloud"
-	etcdCopyBackupsTaskCRDName = "etcdcopybackupstasks.druid.gardener.cloud"
-	crdResources               []string
-)
+// CRDAccess provides methods to manage and get CRDs defined in etcd-druid.
+type CRDAccess interface {
```

Comment:
Can we find a better name to the interface? It is called `CRDAccess` but embeds `component.Deployer`, hence implements all `component.Deployer` funcs like `Deploy`/`Destroy`.

Reply:
Sure can you suggest one?

Reply:
`CRDDeployer` should be concerned only with lifecycle actions. The examples that you have given all return `component.Deployer` which is not what is returned when you create `CRDAccess` as it additionally provides methods to get CRDs (as YAML and as CustomResourceDefinition). So i am not sure if this is the correct terminology to use IMHO.

Reply:
>Isn't the var crdDeployer CRDAccess which is part of the current revision of the PR contradicting itself by definition?

Yes this can be corrected and perhaps was a miss. Once we agree on the final name i can also make this change.

Reply:
Yes this looks better. Will you push this change?

Reply:
Could you share in which test it is needed to deploy and to get the CRD?

Reply:
The other is where its used now (as per this PR) is `test/integration/operator/gardener/care`